docs(mend): document the GitHub.com token in hosted apps (#36110)
renovatebot/renovate: docs/usage/mend-hosted/github-com-token.md

> # Purpose of GitHub.com Token

Reviewer: ```suggestion
# Purpose of github.com Token
```

> Repositories that are NOT hosted on GitHub.com (Cloud) will require a token to access GitHub to retrieve release notes and change logs, plus some package files.

Reviewer: ```suggestion
Users of Mend's Renovate apps on Bitbucket Cloud and Azure DevOps need to provide a github.com token retrieve release notes and changelogs, plus some package files.
```

> If a GitHub.com token is not provided, Renovate will continue to run, but with the following limitations

Reviewer: ```suggestion
If github.com credentials are not provided, Renovate will continue to run, but with the following limitations:
```

> - release notes and change logs might be missing from Renovate pull requests

Reviewer: ```suggestion
- release notes and changelogs will be disabled in Renovate pull requests
```

> - updates will not be provided for packages that are only hosted on GitHub.com

Reviewer: ```suggestion
- lookups for packages hosted on github.com may fail due to rate limiting
```

## Do I need to provide a GitHub.com token?

### GitHub.com (Cloud) users

If your repositories are hosted on GitHub.com (Cloud) and you are using the Renovate App on GitHub, you do not need to provide a token for GitHub.com because the Renovate App already has access to GitHub.

### Bitbucket and Azure DevOps users

> If your repositories are hosted on Bitbucket or Azure DevOps, you will need to provide a GitHub.com token if you want your Renovate CRs/MRs to contain release notes and change logs, or if you want updates on packages that live exclusively on GitHub.com.

Reviewer: ```suggestion
If your repositories are hosted on Bitbucket or Azure DevOps, you will need to provide a GitHub.com token if you want your Renovate PRs to contain release notes and change logs, or if you want updates on packages that live exclusively on GitHub.com.
```

If the GitHub.com token is not provided, a warning will be shown in the Developer Portal and in the Dependency Dashboard.

![Portal warning about GitHub.com](../assets/images/github-token-warning.png)

## How to provide a GitHub.com token

Renovate will accept a GitHub.com token from any GitHub user.
The token requires no special access or permissions.

### Step 1: Acquire a GitHub.com token

1. Log in to GitHub.com (any user)
2. Navigate to [Personal access tokens](https://github.com/settings/tokens) in Developer settings
3. Choose `Generate new token` (either fine-grained or classic)
4. Provide a `Token name` (fine-grained) or `Note` (optional for classic token)
5. Choose the expiration date. (No additional scopes or repository permissions are required.)
6. Press `Generate token`
7. Copy the token when it is presented on the screen

> [!IMPORTANT]
>
> Make sure to copy your personal access token now as you will not be able to see this again.

> ### Step 2: Add the token to the Org/Repo credentials

Reviewer: Isn't this done using GITHUB_COM_TOKEN?
